DS-3534 Update crosstab interaction with automated outlier removal

jrwishart opened the pull request:
* Ensure sample size within each subgroup retained after outlier removal
* Conduct the outlier removal on a subgroup level in importance analysis

jrwishart committed:
DS-3534 Update Interaction with Outlier Removal

* Ensure sample size within each subgroup retained after outlier removal
* Conduct the outlier removal on a subgroup level in importance analysis

jrwishart committed:
DS-3534 Point to dev flipFormat on travis

jrwishart committed:
DS-3534 Point to flipFormat@DS-3534 in DESC file for travis

jrwishart committed:
fixup! DS-3534 Point to flipFormat@DS-3534 in DESC file for travis

mwmclean commented on R/relativeimportance.R:
```diff
@@ -279,13 +283,18 @@ subsetDataWeightsAndFormula <- function(formula, data, weights)
 
+    if (!"non.outlier.data_GQ9KqD7YOf" %in% names(data) &&
+        !is.null(outlier.prop.to.remove) && outlier.prop.to.remove > 0)
+        relevant.data <- removeOutliersFromLinearRegression()
```
`removeDataThatAreOutliersFromLinearRegression(formula, relevant.data, weights,
                                                                       outlier.prop.to.remove = outlier.prop.to.remove)`

jrwishart replied:
Done in https://github.com/Displayr/flipRegression/pull/44/commits/affc993cef376b95cd581e6c613aba41416ef249 and unit test added. Thanks

mwmclean commented:
How about unit tests for `split.size` of the regression output when outlier removal is performed with a crosstab interaction?

jrwishart committed:
DS-3534 Add tests and ensure outlier removal in Jaccard

jrwishart committed:
DS-3534 Ensure ChartData correct

Only name the sample size row when outlier removal not used in
crosstab interaction. The sample size information is already labelled in
this case and the previous code will mislabel it.
